services/fonts.js

```diff
@@ -0,0 +1,225 @@
+    params: pick(config, ['ownerId'])
+  });
+};
+
+/**
+ * Add a font
+ *
+ * See [the corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#add-a-font).
+ *
+ * @param {Object} config
+ * @param {UploadableFile} config.file - A TTF or OTF file.
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.putFont({
+ *   file: 'path/to/file.ttf'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const newFont = response.body;
+ *   });
+ */
+Fonts.putFont = function(config) {
+  v.assertShape({
+    file: v.file,
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'PUT',
+    path: '/fonts/v1/:ownerId',
+    params: pick(config, ['ownerId']),
+    file: config.file
+  });
+};
+
+/**
+ * Delete a font.
+ *
+ * See [the corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#delete-a-font).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.deleteFont({
+ *   font: 'Custom Font Regular'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     // delete successful
+ *   });
+ */
+Fonts.deleteFont = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'DELETE',
+    path: '/fonts/v1/:ownerId/:font',
+    params: config
+  });
+};
+
+/**
+ * Get font metadata.
+ *
+ * See the [corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#retrieve-font-metadata).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.getFontMetadata({
+ *   font: 'Custom Font Regular'
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const metadata = response.body;
+ *   });
+ */
+Fonts.getFontMetadata = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    ownerId: v.string
+  })(config);
+
+  return this.client.createRequest({
+    method: 'GET',
+    path: '/fonts/v1/:ownerId/:font/metadata',
+    params: pick(config, ['ownerId', 'font'])
+  });
+};
+
+/**
+ * Update font metadata.
+ *
+ * See the [corresponding HTTP service documentation](https://docs.mapbox.com/api/maps/fonts/#update-font-metadata).
+ *
+ * @param {Object} config
+ * @param {string} config.font
+ * @param {Object} config.metadata - Font metadata JSON object.
+ * @param {string} [config.ownerId]
+ * @return {MapiRequest}
+ *
+ * @example
+ * fontsClient.updateFontMetadata({
+ *   font: 'Custom Font Regular',
+ *   metadata: {
+ *     visibility: 'public'
+ *   }
+ * })
+ *   .send()
+ *   .then(response => {
+ *     const metadata = response.body;
+ *   });
+ */
+Fonts.updateFontMetadata = function(config) {
+  v.assertShape({
+    font: v.required(v.string),
+    metadata: v.required(v.plainObject),
```

Comment:
I guess that way is more strict as we can validate it more easily. If there were every any other metadata attributes which could be set in the future it might be better to leave them all bundled into a metadata object, but I'll make this change as suggested.